src/vim9execute.c

```diff
@@ -1774,7 +1774,7 @@ ex_disassemble(exarg_T *eap)
 		}
 		break;
 	    case ISN_EXECUTE:
-		smsg("%4d EXECUTE %d", current, iptr->isn_arg.number);
+		smsg("%4d EXECUTE %lld", current, iptr->isn_arg.number);
```

Comment:
Type of `iptr->isn_arg.number` is `varnumber_T`, which
can be either `long`, `int64_t`, or `long long`.
So using `%lld` is not always correct.

More correct would be something like this:
```
smsg("%4d EXECUTE %lld", current, (long long)(iptr->isn_arg.number));
```
I'll try to understand later why we did not detects this earlier
with compiler warning when building with -Wformat=2.

Reply:
I see, the declaration in proto.h is wrong.  After making this change, gcc or clang do complain about the incorrect format:
```
$ git diff proto.h
diff --git a/src/proto.h b/src/proto.h
index a8faa83ca..74f38cd03 100644
--- a/src/proto.h
+++ b/src/proto.h
@@ -119,7 +119,7 @@ extern int _stricoll(char *a, char *b);
 // These prototypes cannot be produced automatically.
 int smsg(const char *, ...)
 # ifdef USE_PRINTF_FORMAT_ATTRIBUTE
-    __attribute__((format(printf, 1, 0)))
+    __attribute__((format(printf, 1, 2)))
 # endif
     ;
 
@@ -138,14 +138,14 @@ int smsg_attr_keep(int, const char *, ...)
 // These prototypes cannot be produced automatically.
 int semsg(const char *, ...)
 # ifdef USE_PRINTF_FORMAT_ATTRIBUTE
-    __attribute__((format(printf, 1, 0)))
+    __attribute__((format(printf, 1, 2)))
 # endif
     ;
 
 // These prototypes cannot be produced automatically.
 void siemsg(const char *, ...)
 # ifdef USE_PRINTF_FORMAT_ATTRIBUTE
-    __attribute__((format(printf, 1, 0)))
+    __attribute__((format(printf, 1, 2)))
 # endif
     ;
```

Then gcc warns about this incorrect format and others:
```
$ gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16   -g -Wall -Wformat=2 -Wno-format-nonliteral -Wextra -Wshadow -Wmissing-prototypes -Wunreachable-code -Wno-deprecated-declarations -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1        -o objects/vim9execute.o vim9execute.c
vim9execute.c: In function ‘ex_disassemble’:
vim9execute.c:1777:22: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘varnumber_T {aka long long int}’ [-Wformat=]
   smsg("%4d EXECUTE %d", current, iptr->isn_arg.number);
                     ~^            ~~~~~~~~~~~~~~~~~~~~
                     %lld
vim9execute.c:1820:23: warning: format ‘%c’ expects argument of type ‘int’, but argument 3 has type ‘varnumber_T {aka long long int}’ [-Wformat=]
   smsg("%4d LOADREG @%c", current, iptr->isn_arg.number);
                      ~^            ~~~~~~~~~~~~~~~~~~~~
                      %lld
vim9execute.c:1865:24: warning: format ‘%c’ expects argument of type ‘int’, but argument 3 has type ‘varnumber_T {aka long long int}’ [-Wformat=]
   smsg("%4d STOREREG @%c", current, iptr->isn_arg.number);
                       ~^            ~~~~~~~~~~~~~~~~~~~~
                       %lld
vim9execute.c:2156:49: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘varnumber_T {aka long long int}’ [-Wformat=]
      case ISN_2STRING: smsg("%4d 2STRING stack[%d]", current,
                                                ~^
                                                %lld
         iptr->isn_arg.number);
         ~~~~~~~~~~~~~~~~~~~~               
```

I also see incorrect formats in other files.  I'll prepare a pull request.

Reply:
See https://github.com/vim/vim/pull/5834 which is a generalization of the fix in this review.